packages/strapi-plugin-content-manager/services/ContentManager.js

```diff
@@ -33,38 +36,50 @@ module.exports = {
 
   async createMultipart(data, { files = {}, model, source } = {}) {
     const entry = await strapi.query(model, source).create(data);
+    const idAttribute = getIdAttribute(model, source);
 
     await uploadFiles(entry, files, { model, source });
 
-    return strapi.query(model, source).findOne({ id: entry.id });
+    return strapi
+      .query(model, source)
+      .findOne({ [idAttribute]: entry[idAttribute] });
   },
 
   async create(data, { files, model, source } = {}) {
     const entry = await strapi.query(model, source).create(data);
+    const idAttribute = getIdAttribute(model, source);
 
     if (files) {
       await uploadFiles(entry, files, { model, source });
-      return strapi.query(model, source).findOne({ id: entry.id });
+      return strapi
+        .query(model, source)
+        .findOne({ [idAttribute]: entry[idAttribute] });
     }
 
     return entry;
   },
 
   async edit(params, data, { model, source, files } = {}) {
+    const idAttribute = getIdAttribute(model, source);
     const entry = await strapi
       .query(model, source)
-      .update({ id: params.id }, data);
+      .update({ [idAttribute]: params.id }, data);
 
     if (files) {
       await uploadFiles(entry, files, { model, source });
-      return strapi.query(model, source).findOne({ id: entry.id });
+      return strapi
+        .query(model, source)
+        .findOne({ [idAttribute]: entry[idAttribute] });
     }
 
     return entry;
   },
 
   delete(params, { source }) {
-    return strapi.query(params.model, source).delete({ id: params.id });
+    const idAttribute = getIdAttribute(params.model, source);
+    return strapi
+      .query(params.model, source)
+      .delete({ [idAttribute]: params.id });
```

Comment:
same for delete

Reply:
do you think it should be changed back to id?

Reply:
Yes leave that to the query layer this allows plugin developpers to ignore this level of complexity

Reply:
ok, sure. Done